packages/cli/src/commands/optimize/optimize.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @flow
+ */
+
+import fs from 'fs';
+import type {ConfigT} from 'types';
+import type {CommandLineArgs} from './optimizeCommandLineArgs';
+import optimizeCommandLineArgs from './optimizeCommandLineArgs';
+import inquirer from 'inquirer';
+import * as AssetUtils from './AssetUtils';
+import {logger} from '@react-native-community/cli-tools';
+import chalk from 'chalk';
+
+async function optimize(_: Array<string>, ctx: ConfigT, args: CommandLineArgs) {
+  const options = {
+    ...(typeof args.save !== 'undefined' ? {save: args.save} : {}),
+    ...(typeof args.quality !== 'undefined' ? {quality: args.quality} : {}),
+    ...(typeof args.include !== 'undefined' ? {include: args.include} : {}),
+    ...(typeof args.exclude !== 'undefined' ? {exclude: args.exclude} : {}),
+  };
+  const hasUnoptimizedAssets = await AssetUtils.hasUnoptimizedAssetsAsync(
+    ctx.root,
+    options,
+  );
+  if (!args.save && hasUnoptimizedAssets) {
+    logger.warn('Running this command will overwrite the original assets.');
+    const {saveOriginals} = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'saveOriginals',
+        message: 'Do you want to save a backup of each file?',
+      },
+    ]);
+    if (saveOriginals) {
+      options.save = true;
+    }
+  }
+  await optimizeAsync(ctx.root, options);
+}
+
+const optimizeAsync = async (projectDir, options = {}) => {
```

Comment:
nit: the `Async` naming convention is Expo convention. While interesting, we don't name our async function this way in the CLI. I'd suggest to rename it to keep the code aligned with the styling.

Reply:
I will change that